File: 1324.竖直打印单词.go
/*
 * @lc app=leetcode.cn id=1324 lang=golang
 *
 * [1324] 竖直打印单词
 */

// @lc code=start
func printVertically(s string) []string {
	words := strings.Split(s, " ")
	maxLen := 0
	for _, word := range words {
		if len(word) > maxLen {
			maxLen = len(word)
		}
	}
	res := make([]string, maxLen)
	for i := 0; i < maxLen; i++ {
		for _, word := range words {
			if i < len(word) {
				res[i] += string(word[i])
			} else {
				res[i] += " "
			}
		}
	}
	for i := 0; i < maxLen; i++ {
		res[i] = strings.TrimRight(res[i], " ")
	}
	return res

}

// @lc code=end

File: 1324.竖直打印单词.py
#
# @lc app=leetcode.cn id=1324 lang=python3
#
# [1324] 竖直打印单词
#

# @lc code=start
class Solution:
    def printVertically(self, s: str) -> List[str]:
        words = s.split()
        max_len = max(map(len, words))
        res = []
        for i in range(max_len):
            res.append(''.join([w[i] if i < len(w) else ' ' for w in words]).rstrip())
        return res
# @lc code=end

